### PR TITLE
Windows support

### PR DESCRIPTION
**Pull request by wjwwood**
This pull request fixes compiler errors and warning for Windows.

This is a work in progress as some C++11 features are not working as advertised with VS 12 2013.

Connects to ros2/ros2#9

### CODE REVIEW

**Comment by tfoote** [comment]
lgtm

File: ros2.repos [hunk 1]
```diff
@@ -2,7 +2,7 @@ repositories:
   ament/ament_cmake:
     type: git
     url: https://github.com/ament/ament_cmake.git
-    version: master
+    version: windows
```
Comment (esteve) [inline ros2.repos]: Does this have to reverted to `master` when all the other PR have been merged?

**Comment by esteve** [comment]
+1

File: ros2.repos (hunk 1 above)
Comment (wjwwood) [inline ros2.repos]: Yes, @tfoote was using this version of the file for the batch CI job.

**Pull request merged**